### PR TITLE
SC-2661: Blackfire && NewRelic Support

### PR DESCRIPTION
**Pull request by alexanderM91**
- Ticket: 
https://spryker.atlassian.net/browse/SC-2661
https://spryker.atlassian.net/browse/SC-2676

- Suite-NonSplit PR:  
https://github.com/spryker/suite-nonsplit/pull/4031

- Release Group:
https://release.spryker.com/release-groups/view/2619

##### Change log:

- Added Blackfire support.
- Added NewRelic support.
- Introduced an ability to configure basic PHP image with enabled extension section.

### CODE REVIEW

File: generator/src/templates/service/blackfire/1.34.0/blackfire.yml.twig [hunk 1]
```diff
@@ -0,0 +1,11 @@
+  {{ serviceName }}:
```
Comment (alexanderM91) [inline generator/src/templates/service/blackfire/1.34.0/blackfire.yml.twig:1]: Please update ReadMe file

File: generator/src/templates/deploy.bash.twig [hunk 2]
```diff
@@ -43,7 +43,7 @@ readonly XARGS_NO_RUN_IF_EMPTY=$(echo '' | xargs echo "--no-run-if-empty")
 # Predefined constants
 readonly APPLICATION_ENV="{{ environment | default('docker') }}"
 declare -a -r SPRYKER_APPLICATIONS=(Glue Yves Zed)
-readonly SPRYKER_PLATFORM_IMAGE=${SPRYKER_PLATFORM_IMAGE:-{{ image | default('spryker/php:7.2') }}}
+readonly SPRYKER_PLATFORM_IMAGE=${SPRYKER_PLATFORM_IMAGE:-{{ image['tag'] | default('spryker/php:7.2') }}}
```
Comment (alexanderM91) [inline generator/src/templates/deploy.bash.twig:46]: Please check that the behavior is BC, e.g end user didn't override the deploy.*.yml and use custom php image

File: generator/src/templates/deploy.bash.twig [hunk 3]
```diff
@@ -145,13 +145,35 @@ readonly USER_GID="1000"
 [ "$1" != 'help' ] && echo -e "*** ${WARN}DEVELOPMENT MODE${NC}"
 DOCKER_EXTRA_OPTIONS+=" -v ${DEPLOYMENT_DIR}/context/php/conf.d/opcache_dev.ini:/usr/local/etc/php/conf.d/opcache.ini:ro"
 DOCKER_EXTRA_OPTIONS+=" -v ${DEPLOYMENT_DIR}/context/php/php.ini:/usr/local/etc/php/php.ini:ro"
+DOCKER_EXTRA_OPTIONS+=" -v ${DEPLOYMENT_DIR}/context/php/conf.d/99-from-deploy-yaml-php.ini:/usr/local/etc/php/conf.d/99-from-deploy-yaml-php.ini:ro"
```
Comment (sprymiker) [inline generator/src/templates/deploy.bash.twig:148]: We do not need it as it already baked in all images

File: images/cli/dev/Dockerfile [hunk 4]
```diff
@@ -4,6 +4,11 @@ ARG SPRYKER_DOCKER_PREFIX
 
 FROM ${SPRYKER_DOCKER_PREFIX}_app:${SPRYKER_DOCKER_TAG}
 
+RUN mkdir -p /tmp/blackfire \
```
Comment (sprymiker) [inline images/cli/dev/Dockerfile:7]: Please, add comment
`# Blackfire client`

File: generator/index.php [hunk 5]
```diff
@@ -450,3 +467,91 @@ function isHost(string $knownHost): bool
...
+    }
+
+    foreach ($projectData['docker']['newrelic'] as $key => $value) {
+        $newrelicEnvVariables['NEWRELIC_' . strtoupper($key)] = $value;
+    }
+
+    return $newrelicEnvVariables;
+}
+
+function buildNewrelicPhpIniConfig(array $projectData): array
```
Comment (sprymiker) [inline generator/index.php:493]: `newrelic.enabled = true'` will override `ENV NEWRELIC_ENABLED=0` in cli image. I assume NEWRELIC_ENABLED=1 (from buildNewrelicEnvVariables) will be enough.

Comment (sprymiker) [reply generator/index.php:493]: Let's remove this method and its usages completely.

File: bin/images/baked.sh [hunk 6]
```diff
@@ -38,6 +38,7 @@ function doBaseImage()
         --build-arg SPRYKER_LOG_DIRECTORY=${logDirectory} \
         --build-arg SPRYKER_COMPOSER_MODE=${SPRYKER_COMPOSER_MODE} \
         --build-arg SPRYKER_COMPOSER_AUTOLOAD=${SPRYKER_COMPOSER_AUTOLOAD} \
+        --build-arg ENABLED_PHP_EXTENSIONS="${ENABLED_PHP_EXTENSIONS}" \
```
Comment (sprymiker) [inline bin/images/baked.sh:41]: We do not need this anymore.

File: bin/images/mount.sh [hunk 7]
```diff
@@ -61,6 +61,7 @@ function buildBaseImages()
         --build-arg SPRYKER_LOG_DIRECTORY=${logDirectory} \
         --build-arg SPRYKER_COMPOSER_MODE=${SPRYKER_COMPOSER_MODE} \
         --build-arg SPRYKER_COMPOSER_AUTOLOAD=${SPRYKER_COMPOSER_AUTOLOAD} \
+        --build-arg ENABLED_PHP_EXTENSIONS="${ENABLED_PHP_EXTENSIONS}" \
```
Comment (sprymiker) [inline bin/images/mount.sh:64]: We do not need this anymore.

File: generator/src/templates/service/blackfire/1.34.0/blackfire.yml.twig [hunk 8]
```diff
@@ -0,0 +1,11 @@
...
+    image: blackfire/blackfire:1.34.0
```
Comment (sprymiker) [inline generator/src/templates/service/blackfire/1.34.0/blackfire.yml.twig:2]: Do we always need exact version. Or can we just use latest?

File: generator/src/templates/images/main/Dockerfile.twig [hunk 9]
```diff
@@ -0,0 +1,15 @@
...
+ARG SPRYKER_PLATFORM_IMAGE=spryker/php:7.2
+FROM ${SPRYKER_PLATFORM_IMAGE} AS spryker_app_dependencies
+
+{% for envName, envValue in _envs %}
+ENV {{ envName }}='{{ envValue }}'
+{% endfor %}
+
+{% if _phpExtensions is defined and _phpExtensions is not empty %}
+{% for phpExtention in _phpExtensions %}
+RUN mv /usr/local/etc/php/disabled/{{phpExtention}}.ini /usr/local/etc/php/conf.d/{{phpExtention}}.ini
```
Comment (sprymiker) [inline generator/src/templates/images/main/Dockerfile.twig:11]: Copy to 90-* to include before custom ini settings from deploy.yml

File: README.md [hunk 10]
```diff
@@ -59,3 +59,4 @@ To start working with Spryker in Docker, follow [the link](https://documentation
 | swagger  | swagger-ui  | v3.24  |
 | kibana  | kibana  | 5.6 |
 |   |   | 6.8 |
+| blackfire  | blackfire  | 1.34.0 |
```
Comment (sprymiker) [inline README.md:62]: latest

**Pull request merged**